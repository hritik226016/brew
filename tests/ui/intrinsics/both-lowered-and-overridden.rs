//! Check that intrinsics that do not get overridden, but are marked as such,
//! cause an error instead of silently invoking the body.
#![feature(intrinsics)]
//@ check-fail
//@ failure-status:101
//@ normalize-stderr-test: ".*note: .*\n\n" -> ""
//@ normalize-stderr-test: "thread 'rustc' panicked.*:\n.*\n" -> ""
//@ normalize-stderr-test: "internal compiler error:.*: " -> ""
//@ error-pattern: intrinsic const_deallocate should be marked with at most one of rustc_intrinsic_must_be_overridden and rustc_intrinsic_lowers_to_mir
//@ rustc-env:RUST_BACKTRACE=0

#[rustc_intrinsic]
#[rustc_intrinsic_lowers_to_mir]
#[rustc_intrinsic_must_be_overridden]
pub const unsafe fn const_deallocate(_ptr: *mut u8, _size: usize, _align: usize) {}

fn main() {
    unsafe { const_deallocate(std::ptr::null_mut(), 0, 0) }
}
