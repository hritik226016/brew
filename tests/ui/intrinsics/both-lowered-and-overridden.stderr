error: intrinsic const_deallocate should be marked with at most one of rustc_intrinsic_must_be_overridden and rustc_intrinsic_lowers_to_mir

query stack during panic:
#0 [intrinsic_raw] fetch intrinsic name if `const_deallocate` is an intrinsic
#1 [check_well_formed] checking that `const_deallocate` is well-formed
end of query stack
error: aborting due to 1 previous error

